### ISSUE TITLE
Upload same file to multiple destinations

### ISSUE THREAD

**Issue by marijap93** [expressjs/multer#453]
Hello guys,

I would like to upload **one image** and then save it to **two different destinations** with two different names. Is there a way to do it?

My code looks like this:
```
var storage = multer.diskStorage({
     destination: function(req, file, cb) {
        cb(null, 'uploads/');
    },
    filename: function(req, file, cb) {
        cb(null, file.originalname);
    }
});

var upload = multer({
    storage: storage,
});

router.post('/api/photo', upload.any(), function(req, res, next) {
    res.send(req.files);

    var path = req.files[0].path;
    var imageName = req.files[0].originalname;
    var author = {
        id: req.user._id,
        username: req.user.username
    };
    var imagepath = {};
    imagepath['path'] = path;
    imagepath['originalname'] = imageName;
    imagepath['author'] = author;
    
    router.addImage(imagepath, function(err) {
    });
    
});
```

My question is, how do I add two different destinations in the **storage** variable? Or is there any other way to do this?

Thank you.

**Comment by marijap93** [comment]
I solved this by copying the file to another destination with a different name, which is exactly what i needed :)
So, i'm closing this issue now.

**Comment by LinusU** [comment]
Glad you resolved it, this will be "fixed" by #399 since the file based storage engine is going away. Then you will get the file as a stream right away, and can store it yourself in an appropriate location :)
